src/toggle-button/styles.scss

```diff
@@ -0,0 +1,18 @@
+/*
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+@use '../internal/styles/tokens' as awsui;
+
+.variant-normal.pressed {
```

Comment:
I have some concerns about relying on selector specificity here. If adding for example a selector like `variant-normal.something-else` somewhere in the button component, specificity will start to get uncertain and hard to manage.

Usually I try to avoid CSS overrides altogether by using the `:not` pseudo-selector, like
```
.variant-normal.pressed {...}
.variant-normal:not(.pressed) {...}
```

Although the `pressed` class doesn't conceptually belong to the Button component (i.e, not Toggle Button) strictly speaking, maybe it is still reasonable to add such pseudo-selectors in there if this avoids bigger refactors.

Reply:
the toggle-button does not have any specific styles for not pressed state. what is the benefit of introducing additional complexity with preudo-selectors?

Reply:
> the toggle-button does not have any specific styles for not pressed state

No but the button component [does](https://github.com/cloudscape-design/components/blob/dbcc7a18b8d7847c4ccc64d38fab6d85684ead42/src/button/styles.scss), that is the problem. And actually I think _we rely_ on subtle CSS specificity calculations because for example for the hover background color, we rely on this selector generated in the button component
```
.button.variant-normal:hover
```

to be more specific than this one generated in the toggle button component:
```
.variant-normal.pressed
```

Anyway, let's do the following: can you make sure that our pipeline integration tests include visual regression tests for hover states? Some of our components already have such tests so it shouldn't be difficult.